pkg/pipeline/transform/transform_network_test.go

```diff
@@ -366,3 +366,147 @@ func (*fakeKubeData) GetInfo(n string) (*kubernetes.Info, error) {
 	}
 	return nil, errors.New("notFound")
 }
+
+func Test_Categorize(t *testing.T) {
+	entryReporterSource := config.GenericMap{
+		"ReporterIP":    "10.1.2.3",
+		"SrcHostIP":     "10.1.2.3",
+		"DstHostIP":     "10.1.2.4",
+		"FlowDirection": "whatever",
+	}
+	entryReporterDest := config.GenericMap{
+		"ReporterIP":    "10.1.2.4",
+		"SrcHostIP":     "10.1.2.3",
+		"DstHostIP":     "10.1.2.4",
+		"FlowDirection": "whatever",
+	}
+	entryReporterUnknown := config.GenericMap{
+		"ReporterIP":    "10.1.2.100",
+		"SrcHostIP":     "10.1.2.3",
+		"DstHostIP":     "10.1.2.4",
+		"FlowDirection": "whatever",
+	}
+	entryReporterSourceNoDest := config.GenericMap{
+		"ReporterIP":    "10.1.2.3",
+		"SrcHostIP":     "10.1.2.3",
+		"FlowDirection": "whatever",
+	}
+	entryReporterDestNoSrc := config.GenericMap{
+		"ReporterIP":    "10.1.2.4",
+		"DstHostIP":     "10.1.2.4",
+		"FlowDirection": "whatever",
+	}
+	entryReporterNoInfo := config.GenericMap{
+		"ReporterIP":    "10.1.2.4",
+		"FlowDirection": "whatever",
+	}
+	entryNoReporter := config.GenericMap{
+		"SrcHostIP":     "10.1.2.3",
+		"FlowDirection": "whatever",
+	}
+	entryNoFlowdir := config.GenericMap{
+		"ReporterIP": "10.1.2.3",
+		"SrcHostIP":  "10.1.2.3",
+		"DstHostIP":  "10.1.2.4",
+	}
+	cfg := config.StageParam{
+		Transform: &config.Transform{
+			Network: &api.TransformNetwork{
+				Rules: []api.NetworkTransformRule{{
+					Type: "reinterpret_direction",
+				}},
+				DirectionInfo: api.DirectionInfo{
+					ReporterIPField:    "ReporterIP",
+					SrcHostField:       "SrcHostIP",
+					DstHostField:       "DstHostIP",
+					FlowDirectionField: "FlowDirection",
+					IfDirectionField:   "IfDirection",
+				},
+			},
+		},
+	}
+
+	tr, err := NewTransformNetwork(cfg)
+	require.NoError(t, err)
+
+	// Source reporter => egress (1)
+	output, ok := tr.Transform(entryReporterSource)
```

Comment:
I would either:
- embed here the struct instantiation so we can see the input and the expected output in the same page, without having to stroll up and down to understand the tests.
or,
- use an array of testcases where you can see them all together, like in https://github.com/netobserv/netobserv-ebpf-agent/pull/78/files#diff-029307d4e44046bf27e28c7ae51e1673c020ff459796821bf11d33d382082e77R34-R72

Reply:
ah, I must say I prefer the flat / straightforward approach rather than the array of test cases, but I get your point of seeing the input and expectations side by side ; i'll update that

Reply:
PS: my rationale for flat test cases is just that I find it easier to quickly jump to a faulty assertion , but that's really a matter of preference